### ISSUE TITLE
Error in Native with Sentry 6

### ISSUE THREAD

**Issue by ia3andy** [quarkiverse/quarkus-logging-sentry#62]
Hello @maciejwalkowiak,

We've got an exception in native when updating to recent Sentry (From 5.x to 6.x):
```
Error: Error parsing reflection configuration in jar:file:/project/lib/io.sentry.sentry-6.1.4.jar!/META-INF/native-image/io.sentry/sentry/reflect-config.json:
Method io.sentry.protocol.SdkVersion.<init>() not found. To allow unresolvable reflection configuration, use option --allow-incomplete-classpath
Verify that the configuration matches the schema described in the -H:PrintFlags=+ output for option ReflectionConfigurationResources.
com.oracle.svm.core.util.UserError$UserException: Error parsing reflection configuration in jar:file:/project/lib/io.sentry.sentry-6.1.4.jar!/META-INF/native-image/io.sentry/sentry/reflect-config.json:
Method io.sentry.protocol.SdkVersion.<init>() not found. To allow unresolvable reflection configuration, use option --allow-incomplete-classpath
```

https://github.com/quarkusio/registry.quarkus.io/pull/113

Any idea?

**Comment by ia3andy** [comment]
cc @gastaldi

**Comment by ia3andy** [comment]
Maybe the fact that `SdkVersion` doesn't have default constructor?

**Comment by ia3andy** [comment]
It's really weird this error is happening randomly (not on all CI runs)

**Comment by ia3andy** [comment]
@gsmet @geoand any thoughts?

**Comment by geoand** [comment]
> Maybe the fact that `SdkVersion` doesn't have default constructor?

The class likely needs to be registered for reflection

**Comment by ia3andy** [comment]
> > Maybe the fact that `SdkVersion` doesn't have default constructor?
> 
> The class likely needs to be registered for reflection

https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/resources/META-INF/native-image/io.sentry/sentry/reflect-config.json#L94

it is

**Comment by ia3andy** [comment]
<img width="1056" alt="image" src="https://user-images.githubusercontent.com/2223984/177717246-d5c84f4a-5162-4128-a48e-b4dca811faf4.png">

**Comment by geoand** [comment]
What does the class look like?

**Comment by gsmet** [comment]
The problem is that it registers specifically a no-args constructor which has been replaced here:
https://github.com/getsentry/sentry-java/commit/f8332e8825e2dfc4c14fd6cd83e1a08151d0f495#diff-f60a611c8c3586d27bd4918b09d6cd0156f4f8337765e3b6c2cf5a424ab92317R59

and then dropped.

The reflection config needs to be updated to point to the new constructor (or the class to be registered globally without being so specific).

**Comment by geoand** [comment]
Which just goes to show you why the json based GraalVM configuration is very brittle

**Comment by ia3andy** [comment]
I've created an upstream issue, thanks guys!

**Comment by maciejwalkowiak** [comment]
@ia3andy thanks for pinging and I am sorry for an issue. I don't work actively anymore on Sentry SDK. Please mention @adinauer in the future.

**Comment by gastaldi** [comment]
> It's really weird this error is happening randomly (not on all CI runs):
> 
> * Passed: https://github.com/quarkiverse/quarkus-logging-sentry/runs/7226494893?check_suite_focus=true
> * Failed: https://github.com/quarkiverse/quarkus-logging-sentry/runs/7085309690?check_suite_focus=true
> * Failed: https://github.com/quarkiverse/quarkus-logging-sentry/runs/6786456191?check_suite_focus=true
> 
> It started failing for Sentry 6.0

That's because they update different dependencies, the ones updating Sentry always fails

**Comment by ia3andy** [comment]
ok it's fixed with 6.3.0, I will release today.
